oidc-service-impl/src/main/java/io/mosip/esignet/services/TokenServiceImpl.java

```diff
@@ -122,6 +124,7 @@ public String getAccessToken(OIDCTransaction transaction, String cNonce) {
         payload.put(CLIENT_ID, transaction.getClientId());
 
         if(cNonce != null) {
+            payload.put(AUD, scopesResourceMapping.getOrDefault(transaction.getPermittedScopes().get(0), ""));
```

Comment:
Why decision making on the resource is very closely tied to the cNonce ? cNonce feature can also be used by open id connect later

Reply:
Done